x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -628,6 +636,61 @@ protected Clock getClock() {
         return components;
     }
 
+    private void generateElasticPassword(SecurityIndexManager.State previousState, SecurityIndexManager.State currentState) {
+        if (previousState.equals(SecurityIndexManager.State.UNRECOVERED_STATE)
+            && currentState.equals(SecurityIndexManager.State.UNRECOVERED_STATE) == false
+            && securityIndex.get().indexExists() == false) {
+
+            final SecureString elasticPassword = new SecureString(generatePassword(20));
+            nativeUsersStoreReference.get()
+                .createReservedUser(
+                    ElasticUser.NAME,
+                    elasticPassword.getChars(),
+                    ActionListener.wrap(
+                        r -> {
```

Comment:
We could pass a reference to the original PrintStream from Bootstrap to Node and then to the SecurityPlugin's `createComponents` so that we can temporarily call `System.setOut` to that , print to stdout and revert to log4j afterwards. This needs `setIO` RuntimePermission for the security plugin though and I'm not sure how comfortable I am with this change.

Reply:
I don't understand why you'd need to call `System.setOut` - you just need a reference to the `PrintStream` so you can use it.

What we need is for `Bootstrap` (or something close to it) to hold on to the original stdout in a place that we can get to it (but is on the forbidden APIs list).

`Bootstrap` already has code to close original stream if we're not in the foreground, so I think a change to add a `SystemConsole` class in `org.elasticsearch.bootstrap` is entirely feasible.

Do you want me to pull something together?

Reply:
> I don't understand why you'd need to call System.setOut - you just need a reference to the PrintStream so you can use it.

Yes, you are right. I misunderstood how we do it ( and how handling stdout works most probably ) in LogConfigurator, and I had presumed the `PrintStream` would be unusable. 

> Do you want me to pull something together?

That would be great